tokio/src/sync/watch.rs

```diff
@@ -10,24 +10,64 @@
 //!
 //! [`channel`] returns a [`Sender`] / [`Receiver`] pair. These are the producer
 //! and consumer halves of the channel. The channel is created with an initial
-//! value. The **latest** value stored in the channel is accessed with
-//! [`Receiver::borrow()`]. Awaiting [`Receiver::changed()`] waits for a new
-//! value to be sent by the [`Sender`] half.
+//! value.
+//!
+//! Each [`Receiver`] independently tracks the last value *seen* by its caller.
+//!
+//! To access the **latest** value stored in the channel and mark it as *seen*,
+//! use [`Receiver::borrow_and_update()`].
+//!
+//! To access the latest value **without** changing its state to *seen*, use
+//! [`Receiver::borrow()`]. (If the value has already been marked *seen*,
+//! [`Receiver::borrow()`] is equivalent to [`Receiver::borrow_and_update()`].)
+//!
+//! ## Change notifications
+//!
+//! The [`Receiver`] half provides an asynchronous [`changed`] method. This
+//! method is ready when a new, *unseen* value is sent via the [`Sender`] half.
+//!
+//! * The [`changed`] method returns `Ok(())` on receiving a new value, or
+//!   `Err(_)` if the [`Sender`] has been closed.
```

Comment:
_very_ tiny nit, not actually important: this returns a public named error type, so we _could_ actually reference it:
```suggestion
//!   `Err(`[`errors::RecvError`]`)` if the [`Sender`] has been closed.
```

Reply:
Done.